### PR TITLE
Improving Classroom Search

### PR DESCRIPTION
**Pull request by luke0malley**
This PR adds building hours, filtering by proximity, building availability, and intends to add classroom availability. Classroom availability could be determined by fetching classroom schedules from the [ClassroomMeetings API](https://dir.api.it.umich.edu/docs/classrooms/1/types/ClassroomMeetings) each day using activejob. This data could then be parsed to determine the periods of time during which a classroom is available. Building hours are from [LSA facilities and operations](https://lsa.umich.edu/lsa/faculty-staff/facilities---operations/building-hours.html). The holidays observed by the university are provided on the [hr website](https://hr.umich.edu/working-u-m/my-employment/holidays-season-days).

These added filters make it easy for users to quickly find available classrooms in buildings that are currently open, and possibly classrooms nearby to them if they choose to enable location services. The outcome is improved usability of MClassrooms.

### CODE REVIEW

File: app/controllers/rooms_controller.rb
Comment (britaumich) [inline app/controllers/rooms_controller.rb:1]: Line 19 is changing the collection of buildings to all buildings. This is not correct. We show only buildings that have classrooms. This is why we need the buidings_ids variable, it has the ids of buildings with classrooms.

File: app/controllers/rooms_controller.rb [hunk 1]
```diff
@@ -37,14 +39,14 @@ def index
...
     @rooms = @rooms.with_all_characteristics(params[:room_characteristics]) if params[:room_characteristics].present?
-    @rooms = @rooms.where('instructional_seating_count >= ?', params[:min_capacity].to_i) if params[:max_capacity].present?
-    @rooms = @rooms.where('instructional_seating_count <= ?', params[:max_capacity].to_i) if params[:max_capacity].present?
-    @rooms = @rooms.where('facility_code_heprod LIKE ? OR UPPER(nickname) LIKE ?', "%#{params[:classroom_name].upcase}%", "%#{params[:classroom_name].upcase}%") if params[:classroom_name].present?
+    @rooms = @rooms.where("instructional_seating_count >= ?", params[:min_capacity].to_i) if params[:max_capacity].present?
+    @rooms = @rooms.where("instructional_seating_count <= ?", params[:max_capacity].to_i) if params[:max_capacity].present?
+    @rooms = @rooms.where("facility_code_heprod LIKE ? OR UPPER(nickname) LIKE ?", "%#{params[:classroom_name].upcase}%", "%#{params[:classroom_name].upcase}%") if params[:classroom_name].present?
 
-    authorize @rooms
+    #authorize @rooms
```
Comment (britaumich) [inline app/controllers/rooms_controller.rb:46]: Why this line is commented?

File: app/controllers/rooms_controller.rb [hunk 2]
```diff
@@ -16,11 +16,13 @@ def index
       @buildings = Building.where(bldrecnbr: buildings_ids, visible: false).order(:name)
     else
       @buildings = Building.where(bldrecnbr: buildings_ids).order(:name)
+      @buildings = Building.all.order(:name)
```
Comment (britaumich) [inline app/controllers/rooms_controller.rb:19]: Line 19 is changing the collection of buildings to all buildings. This is not correct. We show only buildings that have classrooms. This is why we need the buidings_ids variable, it has the ids of buildings with classrooms.

File: app/views/rooms/index.html.erb [hunk 3]
```diff
@@ -99,23 +141,22 @@
...
+              </div>
             </div>
           </div>
-        </div>
-        <div class="flex-1 mx-2">
-          <div class="" id="rooms">
-            <% if user_signed_in? %>
+          <div class="flex-1 mx-2">
+            <div class="" id="rooms">
               <%= render 'listing' %>
```
Comment (britaumich) [inline app/views/rooms/index.html.erb:154]: Why is this line <% if user_signed_in? %>  removed?

**Comment by rsmoke** [comment]
Rick Smoke closed this due to inactivity. We will re open when we refactor the application.

**Pull request closed without merging**